src/bundle/Resources/public/js/scripts/core/adaptive.items.js

```diff
@@ -0,0 +1,60 @@
+(function (global, doc, eZ) {
+    const OFFSET_ROUNDING_COMPENSATOR = 0.5;
+
+    class AdaptiveItems {
+        constructor(config) {
+            this.items = config.items;
+            this.selectorItem = config.selectorItem;
+            this.itemHiddenClass = config.itemHiddenClass;
+            this.container = config.container;
+            this.getActiveItem = config.getActiveItem;
+
+            this.onAdapted = config.onAdapted;
+        }
+
+        adapt() {
+            [this.selectorItem, ...this.items].forEach((item) => item.classList.remove(this.itemHiddenClass));
+
+            const activeItem = this.getActiveItem();
+            const activeItemWidth = activeItem ? activeItem.offsetWidth + OFFSET_ROUNDING_COMPENSATOR : 0;
+            const selectorWidth = this.selectorItem.offsetWidth + OFFSET_ROUNDING_COMPENSATOR;
+            const maxTotalWidth = this.container.offsetWidth - OFFSET_ROUNDING_COMPENSATOR;
+            const hiddenItemsWithoutSelector = new Set();
+            let currentWidth = selectorWidth + activeItemWidth;
+
+            for (let i = 0; i < this.items.length; i++) {
+                const item = this.items[i];
+
+                if (item === activeItem) {
+                    continue;
+                }
+
+                const lastItem = this.items[this.items.length - 1]
+                const isLastNonactiveItem = lastItem === activeItem ? i === this.items.length - 2 : i === this.items.length - 1;
```

Comment:
what would you say for `this.items.length - (lastItem === activeItem ? 2 : 1)`?

Reply:
So, you are proposing this?
```suggestion
                const isLastNonactiveItem = i === this.items.length - (lastItem === activeItem ? 2 : 1);
```
It is shorter, but for me my version is easier to understand. 
Do you think that your version is easier to understand?

Reply:
I don't. ;) But also I don't like using comparison inside ternary operator, now you have three `===` in one line